checker/src/main/java/org/checkerframework/checker/mustcall/MustCallAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,363 @@
+   * @param type the method or constructor's type
+   */
+  private void changeNonOwningParameterTypesToTop(
+      ExecutableElement declaration, AnnotatedExecutableType type) {
+    for (int i = 0; i < type.getParameterTypes().size(); i++) {
+      Element paramDecl = declaration.getParameters().get(i);
+      if (checker.hasOption(MustCallChecker.NO_LIGHTWEIGHT_OWNERSHIP)
+          || getDeclAnnotation(paramDecl, Owning.class) == null) {
+        AnnotatedTypeMirror paramType = type.getParameterTypes().get(i);
+        if (!paramType.hasAnnotation(POLY)) {
+          paramType.replaceAnnotation(TOP);
+        }
+        if (declaration.isVarArgs() && i == type.getParameterTypes().size() - 1) {
+          // also modify the component type of a varargs array
+          AnnotatedTypeMirror varargsType = ((AnnotatedArrayType) paramType).getComponentType();
+          if (!varargsType.hasAnnotation(POLY)) {
+            varargsType.replaceAnnotation(TOP);
+          }
+        }
+      }
+    }
+  }
+
+  @Override
+  public AnnotatedTypeMirror fromElement(Element elt) {
+    AnnotatedTypeMirror type = super.fromElement(elt);
+    // Support @InheritableMustCall meaning @MustCall on all class declaration elements.
+    if (ElementUtils.isTypeElement(elt)) {
+      AnnotationMirror inheritableMustCall = getDeclAnnotation(elt, InheritableMustCall.class);
+      if (inheritableMustCall != null) {
+        List<String> mustCallVal =
+            AnnotationUtils.getElementValueArray(
+                inheritableMustCall, inheritableMustCallValueElement, String.class);
+        AnnotationMirror inheritedMCAnno = createMustCall(mustCallVal.toArray(new String[0]));
+        // Ensure that there isn't an inconsistent, user-written @MustCall annotation and
+        // issue an error if there is. Otherwise, replace the implicit @MustCall({}) with
+        // the inherited must-call annotation.
+        AnnotationMirror writtenMCAnno = type.getAnnotationInHierarchy(this.TOP);
+        if (writtenMCAnno != null
+            && !this.getQualifierHierarchy().isSubtype(inheritedMCAnno, writtenMCAnno)) {
+          if (!elementsIssuedInconsistentMustCallSubtypeErrors.contains(elt)
+              && !this.checker.shouldSkipUses(elt)) {
+            checker.reportError(
+                elt,
+                "inconsistent.mustcall.subtype",
+                elt.getSimpleName(),
+                writtenMCAnno,
+                inheritableMustCall);
+            elementsIssuedInconsistentMustCallSubtypeErrors.add(elt);
+          }
+        } else {
+          type.replaceAnnotation(inheritedMCAnno);
+        }
+      }
+    }
+    return type;
+  }
+
+  /**
+   * Creates a {@link MustCall} annotation whose values are the given strings.
+   *
+   * @param val the methods that should be called
+   * @return an annotation indicating that the given methods should be called
+   */
+  public AnnotationMirror createMustCall(final String... val) {
```

Comment:
I suspect that a given program will contain very few distinct MustCall annotations.  I suggest renaming this method to a private helper method, and having a public method that uses a cache.